js/FileDrop.js

```diff
@@ -40,16 +40,32 @@ profile, committeesize, _, _ = fileio.read_abcvoting_yaml_file(filetext)
 return_object = {'num_cand': profile.num_cand, 'num_voter': len(profile), 'committeesize': committeesize}
 u = {j : {i : 0 for i in range(len(profile))} for j in range(profile.num_cand)}
 for i, voter in enumerate(profile):
-for candidate in voter.approved:
-u[candidate][i] = 1
+    for candidate in voter.approved:
+        u[candidate][i] = 1
 return_object['u'] = u
+if profile.total_weight() > len(profile):
```

Comment:
Since weights can be fractional and therefore < 1, it would be better to check if any voter has weight != 1, or while in python to just use [profile.has_unit_weights()](https://abcvoting.readthedocs.io/en/latest/preferences.html#abcvoting.preferences.Profile.has_unit_weights)